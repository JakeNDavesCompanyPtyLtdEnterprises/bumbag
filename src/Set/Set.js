// @flow
import React, { type Node } from 'react';

import _Set from './styled';

type Props = {
  className?: string,
  children?: Node,
  spacing?: boolean
};

const Set = ({ children, ...props }: Props) => <_Set {...props}>{children}</_Set>;

Set.defaultProps = {
  className: undefined,
  children: undefined,
  spacing: 'xxsmall'
};

export default Set;
