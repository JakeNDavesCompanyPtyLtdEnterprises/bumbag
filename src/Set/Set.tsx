import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { InlineFlexProps as ReakitInlineFlexProps } from 'reakit/ts/InlineFlex/InlineFlex';

import _Set from './styled';

export interface LocalSetProps {
  className?: string;
  children: React.ReactNode;
  isVertical?: boolean;
  spacing?: string;
}
export type SetProps = ReakitInlineFlexProps & LocalSetProps;

export const Set: React.FunctionComponent<LocalSetProps> = ({ children, ...props }) => (
  <_Set {...props}>{children}</_Set>
);

Set.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  isVertical: PropTypes.bool,
  spacing: PropTypes.string
};
Set.defaultProps = {
  className: undefined,
  isVertical: false,
  spacing: 'xxsmall'
};

// @ts-ignore
const C: React.FunctionComponent<SetProps> = Set;
export default C;
