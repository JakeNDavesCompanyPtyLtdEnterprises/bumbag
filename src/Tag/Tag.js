// @flow
import React, { type Node } from 'react';

import type { Palette, Size } from '../typesold';
import _Tag from './styled';

type Props = {
  className?: string,
  children: Node,
  palette?: Palette,
  size?: Size
};

const Tag = ({ children, ...props }: Props) => <_Tag {...props}>{children}</_Tag>;

Tag.defaultProps = {
  className: undefined,
  palette: 'text',
  title: undefined,
  size: undefined
};

export default Tag;
