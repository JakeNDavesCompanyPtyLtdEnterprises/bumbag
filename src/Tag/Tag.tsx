import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { InlineBlockProps as ReakitInlineBlockProps } from 'reakit/ts/InlineBlock/InlineBlock';

import _Tag from './styled';

export interface LocalTagProps {
  className?: string;
  children: React.ReactNode;
  kind?: 'outlined';
  palette?: string;
  size?: string;
}
export type TagProps = ReakitInlineBlockProps & LocalTagProps;

export const Tag: React.FunctionComponent<LocalTagProps> = ({ children, ...props }) => (
  <_Tag {...props}>{children}</_Tag>
);

Tag.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  kind: PropTypes.oneOf(['outlined']),
  palette: PropTypes.string,
  size: PropTypes.string
};
Tag.defaultProps = {
  className: undefined,
  kind: undefined,
  palette: 'text',
  size: undefined
};

// @ts-ignore
const C: React.FunctionComponent<TagProps> = Tag;
export default Tag;
