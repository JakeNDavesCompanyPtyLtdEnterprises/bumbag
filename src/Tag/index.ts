export { default as Tag } from './Tag';
export { default } from './Tag';
