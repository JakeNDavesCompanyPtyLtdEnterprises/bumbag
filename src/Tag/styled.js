import styled, { css } from 'reakit/styled';
import InlineBlock from 'reakit/InlineBlock';
import { palette, theme } from 'styled-tools';

const sizeAttributes = {
  medium: css`
    font-size: 1em;

    & {
      ${theme('fannypack.Tag.sizes.medium')};
    }
  `,
  large: css`
    font-size: 1.25em;

    & {
      ${theme('fannypack.Tag.sizes.large')};
    }
  `
};

const Tag = styled(InlineBlock)`
  background-color: ${palette()};
  border-radius: 4px;
  color: ${props => palette(`${props.palette}Inverted`)};
  font-size: ${theme('fannypack.fontSizes.small')}rem;
  font-weight: ${theme('fannypack.fontWeights.semibold')};
  padding: 0.2em 0.4em;

  & {
    ${props => props.size && sizeAttributes[props.size]};
  }

  & {
    ${theme('fannypack.Tag.base')};
  }
`;

export default Tag;
