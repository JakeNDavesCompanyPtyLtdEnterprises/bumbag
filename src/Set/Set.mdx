---
name: Set
route: /components/Set
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Set from '../Set';
import Button from '../Button';
import Tag from '../Tag';

# Set

## Import

`import { Set } from 'fannypack'`

## Basic Usage

A `<Set>` can be used to list a set of components. It will automatically space out the components, and wrap them when they reach the width of the viewport.

<Playground>
  <Set>
    <Button>Button 1</Button>
    <Button>Button 2</Button>
    <Button>Button 3</Button>
    <Button>Button 4</Button>
    <Button>Button 5</Button>
    <Button>Button 6</Button>
    <Button>Button 7</Button>
  </Set>
</Playground>

## Spacing

You can modify the spacing of the set with the `spacing` prop.

<Playground>
  <Set spacing="xxxsmall">
    <Tag>Tag 1</Tag>
    <Tag>Tag 2</Tag>
    <Tag>Tag 3</Tag>
    <Tag>Tag 4</Tag>
    <Tag>Tag 5</Tag>
    <Tag>Tag 6</Tag>
    <Tag>Tag 7</Tag>
  </Set>
</Playground>

## Vertical

You can also have a vertical set with the `isVertical` prop.

<Playground>
  <Set isVertical spacing="xxxsmall">
    <Tag>Tag 1</Tag>
    <Tag>Tag 2</Tag>
    <Tag>Tag 3</Tag>
    <Tag>Tag 4</Tag>
    <Tag>Tag 5</Tag>
    <Tag>Tag 6</Tag>
    <Tag>Tag 7</Tag>
  </Set>
</Playground>

## Props

<PropsTable of={Set} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  child: string | Object
}
```
