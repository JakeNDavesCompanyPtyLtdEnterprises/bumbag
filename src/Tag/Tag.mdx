---
name: Tag
route: /components/tag
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Tag from '../Tag';
import Group from '../Group';
import Set from '../Set';

# Tag

## Import

`import { Tag } from 'fannypack'`

## Basic Usage

<Playground>
  <Tag>Default</Tag>
</Playground>

## Colors

<Playground>
  <Set spacing="xxxsmall">
    <Tag palette="primary">Primary</Tag>
    <Tag palette="secondary">Secondary</Tag>
    <Tag palette="success">Success</Tag>
    <Tag palette="warning">Warning</Tag>
    <Tag palette="danger">Danger</Tag>
  </Set>
</Playground>

### Tints

<Playground>
  <Set spacing="xxxsmall">
    <Tag palette="textTint">Default</Tag>
    <Tag palette="primaryTint">Primary</Tag>
    <Tag palette="secondaryTint">Secondary</Tag>
    <Tag palette="successTint">Success</Tag>
    <Tag palette="warningTint">Warning</Tag>
    <Tag palette="dangerTint">Danger</Tag>
  </Set>
</Playground>

## Outlines

An outline can be placed on a tag by assigning `"outlined"` to the `type` prop.

<Playground>
  <Set spacing="xxxsmall">
    <Tag kind="outlined">Default</Tag>
    <Tag palette="primary" kind="outlined">Primary</Tag>
    <Tag palette="secondary" kind="outlined">Secondary</Tag>
    <Tag palette="success" kind="outlined">Success</Tag>
    <Tag palette="warning" kind="outlined">Warning</Tag>
    <Tag palette="danger" kind="outlined">Danger</Tag>
  </Set>
</Playground>

## Sizes

<Playground>
  <Set spacing="xxxsmall">
    <Tag>Default</Tag>
    <Tag size="medium">Medium</Tag>
    <Tag size="large">Large</Tag>
  </Set>
</Playground>

## Sets

To group a set of tags, use the [`<Set>` component](/components/set).

<Playground>
  <Set spacing="xxxsmall">
    <Tag>Hello</Tag>
    <Tag>World</Tag>
    <Tag>This</Tag>
    <Tag>Is</Tag>
    <Tag>Me</Tag>
  </Set>
</Playground>

## Grouping

You can group tags to make one tag with the `<Group>` component.

<Playground>
  <Group>
    <Tag>Default</Tag>
    <Tag palette="textTint">Hello</Tag>
  </Group>
</Playground>

## `<Tag>` Props

<PropsTable of={Tag} />

## Theming

### `<Tag>` Schema

```jsx
{
  base: string | Object,
  sizes: {
    medium: string | Object,
    large: string | Object
  }
}
```
