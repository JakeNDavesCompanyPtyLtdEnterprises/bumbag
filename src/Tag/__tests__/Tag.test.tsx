import * as React from 'react';
import render from '../../_utils/tests/render';
import Tag from '../Tag';
import Set from '../../Set';

it('renders correctly', () => {
  const { container } = render(<Tag>Tag</Tag>);
  expect(container.firstChild).toMatchSnapshot();
});

describe('colors', () => {
  ['danger', 'success', 'warning', 'primary'].forEach(color => {
    it(`renders correctly for a tag with color ${color}`, () => {
      const { container } = render(<Tag palette={color}>Test</Tag>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('sizes', () => {
  ['medium', 'large'].forEach(size => {
    it(`renders correctly for a tag with size ${size}`, () => {
      const { container } = render(<Tag size={size}>Test</Tag>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

it('renders correctly for a list of tags', () => {
  const { container } = render(
    <Set>
      <Tag>Tag 1</Tag>
      <Tag>Tag 2</Tag>
      <Tag>Tag 3</Tag>
    </Set>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an outlined tag', () => {
  const { container } = render(<Tag kind="outlined">Outlined tag</Tag>);
  expect(container.firstChild).toMatchSnapshot();
});
