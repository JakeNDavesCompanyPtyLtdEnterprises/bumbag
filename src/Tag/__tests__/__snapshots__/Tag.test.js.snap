// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors renders correctly for a tag with color danger 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #c9444d;
}

.c0 {
  display: inline-block;
  background-color: #c9444d;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
/>
`;

exports[`colors renders correctly for a tag with color primary 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #444bc9;
}

.c0 {
  display: inline-block;
  background-color: #444bc9;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
/>
`;

exports[`colors renders correctly for a tag with color success 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #2c8453;
}

.c0 {
  display: inline-block;
  background-color: #2c8453;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
/>
`;

exports[`colors renders correctly for a tag with color warning 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #f2a100;
}

.c0 {
  display: inline-block;
  background-color: #f2a100;
  border-radius: 4px;
  color: #000;
  fill: #000;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
/>
`;

exports[`renders correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  display: inline-block;
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

<div
  class="c0 c1"
>
  Tag
</div>
`;

exports[`renders correctly for a list of tags 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2 {
  display: inline-block;
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c0 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
  spacing="xxsmall"
>
  <div
    class="c2 c3"
  >
    Tag 1
  </div>
  <div
    class="c2 c3"
  >
    Tag 2
  </div>
  <div
    class="c2 c3"
  >
    Tag 3
  </div>
</div>
`;

exports[`renders correctly for an outlined tag 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  display: inline-block;
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

.c0 {
  background-color: unset;
  border: 1px solid #435a6f;
  color: #435a6f;
  fill: #435a6f;
}

<div
  class="c0 c1"
  kind="outlined"
>
  Outlined tag
</div>
`;

exports[`sizes renders correctly for a tag with size large 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  display: inline-block;
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

.c0 {
  font-size: 1.25em;
}

<div
  class="c0 c1"
  style="size: large;"
/>
`;

exports[`sizes renders correctly for a tag with size medium 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #435a6f;
}

.c0 {
  display: inline-block;
  background-color: #435a6f;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  font-size: 0.8rem;
  font-weight: 600;
  padding: 0.2em 0.4em;
}

.c0 {
  font-size: 1em;
}

<div
  class="c0 c1"
  style="size: medium;"
/>
`;
